end_to_end/test_decode_checkpoint.sh

```diff
@@ -58,7 +58,7 @@ echo
 $cmd python3 MaxText/train.py MaxText/configs/base.yml \
 run_name=${training_ckpt_run_id} \
 base_output_directory=${base_output_directory} \
-dataset_path=${dataset_path} \
+dataset_path=${dataset_path} attention=mha \
```

Comment:
Again, we need to test TPU limits on TPU and GPU limits on GPU. An interesting option for you would be to add a flag called "recommended" which chooses whatever we recommend and is platform aware.

Reply:
I understand your point but I still think choosing mha over flash doesn't affect what this test is testing. But adding something like `attention=auto` sounds good to me.

Reply:
Ok looks like we're aligned but I think we should just support `attention=mha` and `attention=flash` for now. And `attention=flash` can dispatch to TPU/GPU as needed.